scheduler/generic_sched.go

```diff
@@ -789,7 +789,8 @@ func updateRescheduleTracker(alloc *structs.Allocation, prev *structs.Allocation
 
 // findPreferredNode finds the preferred node for an allocation
 func (s *GenericScheduler) findPreferredNode(place placementResult) (*structs.Node, error) {
-	if prev := place.PreviousAllocation(); prev != nil && place.TaskGroup().EphemeralDisk.Sticky {
+	if prev := place.PreviousAllocation(); prev != nil &&
```

Comment:
I wouldn't mind breaking this up into multiple `if`s with an early return for things like `prev == nil`, but nbd.